main.go

```diff
@@ -77,6 +105,14 @@ func main() {
 		log.Fatalf("Failed to create the clientset with the given config with %v", err)
 	}
 
+	if action == Install {
+		installAction(ctx, cancel, clientSet, config, hr, interval)
+	} else if action == Delete {
+		deleteAction(ctx, cancel, clientSet, hr, timeout, interval)
+	}
+}
+
+func installAction(ctx context.Context, cancel context.CancelFunc, clientSet client.Client, config *rest.Config, hr *fluxhelmv2beta1.HelmRelease, interval time.Duration) {
 	ns := &corev1.Namespace{
```

Comment:
We should call `defer cancel()` at the beginning of this `func` as well

Reply:
Yeah, but lint gives error because deferred functions are not called on os.Exit() which log.Fatalf() does. The existing code path is safe because defer cancel() is done in PollStatus which is always in the exit path (except fatal exits).